docker-compose.yml

```diff
@@ -0,0 +1,7 @@
+app:
+  volumes:
+    - "./:/app"
+    - "/app/node_modules/"
```

Comment:
One problem that @ifsimicoded ran into when she tried docker setup on her local machine was it wouldn't work unless she had prerun `yarn install` before any of the `docker-compose` commands. She discovered this trick on line 4 that apparently prevents the node_modules directory in the docker container from getting overwritten or blown away when the volume referencing happens. 🤷‍♂️

Reply:
The downside to this we've now discovered is that whenever you add new dependencies in `package.json` you have to rebuild your image. To do this, we are running `docker-compose build --no-cache`. Is there a better way to set this up in docker?

Reply:
> There seems to be an important order of operations on copy over files into the volume and running yarn install

The `Dockerfile` gets compiled (which [runs `yarn`](https://github.com/cityofaustin/janis/pull/9/files#diff-3254677a7917c6c01f55212f86c57fbfR5)). So the compiled docker image is good. Then running `docker-compose up` or `docker-compose run` will mount the volumes. In this case you're mounting the whole local directory `.` to the `/app` directory inside the container. So unless you've run `yarn` locally you won't have any `node_modules`.

> She discovered this trick

🎉 nice work @ifsimicoded! This will create a persistent volume that ensures `node_modules` doesn't get mounted from the local filesystem. This seems like an elegant solution.

> Is there a better way to set this up in docker?

Running `docker-compose build --no-cache` seems like the right thing if `docker-compose up --build` doesn't work correctly. I'd say don't use `docker-compose` until you need to. I'd also suggest a `serve-local.sh` or similar command that people can just run without needing to remember the secret incantation from project to project.